src/components/board/client.ts

```diff
@@ -138,7 +138,13 @@ export class BoardClient implements Board {
       pb.ReadAnalogReaderRequest,
       pb.ReadAnalogReaderResponse
     >(boardService.readAnalogReader.bind(boardService), request);
-    return response.getValue();
+    const value: AnalogValue = {
+      value: response.getValue(),
+      minRange: response.getMinRange(),
+      maxRange: response.getMaxRange(),
+      stepSize: response.getStepSize(),
+    };
+    return value;
```

Comment:
can we not just return `response.asObject()` here? (or whatever converts a `pb.ReadAnalogReaderResponse` into a JS Object?)

Reply:
I dont think so, just tried and got `Unsafe return of an `any` typed value.` error

Reply:
Did you also try `return response.toObject()`?

Reply:
it says thats not a function

Reply:
That's... weird. Sorry, I've been spending the last 10 minutes trying to figure out why it would say it's not a function... 

I tried it on my end. (Forgot I had to `make all` first to get the latest proto changes). I also switched the `board.ts` to return a `Promise<AnalogValue>;`.
and changed this to: `return response.toObject();`. It worked for me

Reply:
<img width="800" alt="image" src="https://github.com/viamrobotics/viam-typescript-sdk/assets/31713368/1a70a733-f2a5-4535-9281-b0820510a611">
<img width="327" alt="image" src="https://github.com/viamrobotics/viam-typescript-sdk/assets/31713368/30cd0368-b4cf-49ef-9b44-ed672a8e873e">
I don't know if this helps, but these are my changes. I ran `make format lint` without a problem.

Reply:
Did you run the test? the test fails with the error `AssertionError: promise rejected "TypeError: response.toObject is not a function instead of resolving`

Reply:
Ohhhh yes, that makes sense! Because in your test, you emulate what the other `get<>` functions are, but NOT `toObject()`. So you would have to change the mock to be like something similar to this:
```
  BoardServiceClient.prototype.readAnalogReader = vi
    .fn()
    .mockImplementation((_req, _md, cb) => {
      cb(null, { toObject: () => testAnalogValue });
    });
```

Reply:
The thing you're mocking is the actual **pb.response** type itself. So you have to manually define what the functions of your mock response type would return. Now that the function `readAnalogReader()` is returning `response.toObject()` instead of all the other get functions, that's what you have to mock instead.
Let me know if you don't get it! It took me a while to wrap my head around it too.

Reply:
ohhhh okay i see, that worked thanks so much!